x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -131,15 +193,25 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 			// }
 			zone = strings.TrimSuffix(zone, "*")
 			f = fmt.Sprintf("%s AND resource.label.location=starts_with(\"%s\")", f, zone)
+			break
+		}
+
+		if len(r.config.Regions) != 0 {
+			regionsFilter := r.buildRegionsFilter(r.config.Regions, gcp.GKEResourceLabelLocation)
+			f = fmt.Sprintf("%s AND %s", f, regionsFilter)
 		}
 	case gcp.ServicePubsub, gcp.ServiceLoadBalancing, gcp.ServiceCloudFunctions, gcp.ServiceFirestore, gcp.ServiceDataproc:
 		return f
 	case gcp.ServiceStorage:
-		if r.config.Region == "" {
-			return f
+		switch {
```

Comment:
Here should be added the warning about `region` taking precedence over `regions`